pkg/resources/shared_database.go

```diff
@@ -0,0 +1,266 @@
+		Description:   "A shared database creates a database from a share provided by another Snowflake account. For more information about shares, see [Introduction to Secure Data Sharing](https://docs.snowflake.com/en/user-guide/data-sharing-intro).",
+
+		Schema: sharedDatabaseSchema,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+	}
+}
+
+func CreateSharedDatabase(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+	client := meta.(*provider.Context).Client
+
+	id := sdk.NewAccountObjectIdentifier(d.Get("name").(string))
+	externalShareId := sdk.NewExternalObjectIdentifierFromFullyQualifiedName(d.Get("from_share").(string))
+
+	var externalVolume *sdk.AccountObjectIdentifier
+	if v, ok := d.GetOk("external_volume"); ok {
+		externalVolume = sdk.Pointer(sdk.NewAccountObjectIdentifier(v.(string)))
+	}
+
+	var catalog *sdk.AccountObjectIdentifier
+	if v, ok := d.GetOk("catalog"); ok {
+		catalog = sdk.Pointer(sdk.NewAccountObjectIdentifier(v.(string)))
+	}
+
+	var logLevel *sdk.LogLevel
+	if v, ok := d.GetOk("log_level"); ok {
+		logLevel = sdk.Pointer(sdk.LogLevel(v.(string)))
+	}
+
+	var traceLevel *sdk.TraceLevel
+	if v, ok := d.GetOk("trace_level"); ok {
+		traceLevel = sdk.Pointer(sdk.TraceLevel(v.(string)))
+	}
+
+	err := client.Databases.CreateShared(ctx, id, externalShareId, &sdk.CreateSharedDatabaseOptions{
+		Transient:           GetPropertyAsPointer[bool](d, "is_transient"),
+		ExternalVolume:      externalVolume,
+		Catalog:             catalog,
+		DefaultDDLCollation: GetPropertyAsPointer[string](d, "default_ddl_collation"),
+		LogLevel:            logLevel,
+		TraceLevel:          traceLevel,
+		Comment:             GetPropertyAsPointer[string](d, "comment"),
+	})
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(helpers.EncodeSnowflakeID(id))
+
+	return ReadSharedDatabase(ctx, d, meta)
+}
+
+func UpdateSharedDatabase(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+	client := meta.(*provider.Context).Client
+	id := helpers.DecodeSnowflakeID(d.Id()).(sdk.AccountObjectIdentifier)
+
+	if d.HasChange("name") {
+		newName := sdk.NewAccountObjectIdentifier(d.Get("name").(string))
+		err := client.Databases.Alter(ctx, id, &sdk.AlterDatabaseOptions{
+			NewName: &newName,
+		})
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		d.SetId(helpers.EncodeSnowflakeID(newName))
+		id = newName
+	}
+
+	if d.HasChange("comment") {
+		comment := d.Get("comment").(string)
+		if len(comment) > 0 {
+			err := client.Databases.Alter(ctx, id, &sdk.AlterDatabaseOptions{
+				Set: &sdk.DatabaseSet{
+					Comment: &comment,
+				},
+			})
+			if err != nil {
+				return diag.FromErr(err)
+			}
+		} else {
+			err := client.Databases.Alter(ctx, id, &sdk.AlterDatabaseOptions{
+				Unset: &sdk.DatabaseUnset{
+					Comment: sdk.Bool(true),
+				},
+			})
+			if err != nil {
+				return diag.FromErr(err)
+			}
+		}
+	}
```

Comment:
Missing updating otther fields

Reply:
Only name and comment can be changed. Forgot to mention it in the pr description (I knew I forgot something). Shared objects are immutable, but I found out that name and comment can be changed ([doc reference](https://docs.snowflake.com/en/user-guide/data-sharing-intro)), that's why other parameters/fields are marked as ForceNew.